src/lepton/jpgcoder.cc

```diff
@@ -2003,9 +2003,16 @@ void process_file(IOUtil::FileReader* reader,
     }
     if (!fast_exit) {
         // close iostreams
-        if ( str_in  != NULL ) delete( str_in  ); str_in  = NULL;
-        if ( str_out != NULL ) delete( str_out ); str_out = NULL;
-        if ( ujg_out != NULL ) delete( ujg_out ); ujg_out = NULL;
+        if (ujg_base_in != str_in) {
+            // ujg_base_in may be the same as str_in or Sirikata::BufferedReader(str_in)
+            // Thus we need to delete ujg_base_in, but only if it differs from str_in.
```

Comment:
Why? Delete of null works correctly.

Reply:
yeah, and i systematically remove these checks when refactoring code

Reply:
Pleas look carefully. I don't check for NULL. It was in the old code I replaced.

Reply:
I rechecked the code and actually it's opposite to the comment:
> ujg_base_in may be the same as str_in or Sirikata::BufferedReader(str_in)

should say

> str_in may be the same as ujg_base_in or Sirikata::BufferedReader(ujg_base_in)

While BufferedReader doesn't own its base stream and its destructor is just empty, it will look more correct to destroy str_in first.